hs_core/tests/api/native/test_irods_user_zone_federation.py

```diff
@@ -230,3 +230,36 @@ def test_resource_operations_in_user_zone(self):
 
         # delete resources to clean up
         resource.delete_resource(self.res.short_id)
+
+        # test adding files from federated user zone to an empty resource
+        # created in hydroshare zone
+        res = resource.create_resource(
+            resource_type='GenericResource',
+            owner=self.user,
+            title='My Test Generic Resource in HydroShare Zone'
+        )
+        self.assertEqual(res.files.all().count(), 0,
+                         msg="Number of content files is not equal to 0")
+        fed_test_file1_full_path = '/{zone}/home/testuser/{fname}'.format(
+            zone=settings.HS_USER_IRODS_ZONE, fname=self.file_one)
+        hydroshare.add_resource_files(
+            res.short_id,
+            fed_res_file_names=[fed_test_file1_full_path],
+            fed_copy_or_move='copy')
+        # test resource has one file
+        self.assertEqual(res.files.all().count(), 1,
+                         msg="Number of content files is not equal to 1")
+
+        file_list = []
+        for f in res.files.all():
+            file_list.append(f.fed_resource_file_name_or_path.split('/')[-1])
+        self.assertTrue(self.file_one in file_list,
+                        msg='file 1 has not been added in the resource in hydroshare zone')
+        # test original file in user test zone still exist after adding it to the resource
+        # since 'copy' is used for fed_copy_or_move when adding the file to the resource
+        self.assertTrue(self.irods_storage.exists(user_path + self.file_one))

```

Comment:
@pkdash Although our current code does not mess with user's scratch space after files become resource's content files, it won't hurt to add this check in just in case. I will add this check in after the resource gets deleted then.